vignettes/TechnicalDocmentation.Rmd

```diff
@@ -0,0 +1,636 @@
+---
+title: "Technical Documentation"
+header-includes:
+output: html_document
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = FALSE,message = FALSE)
+```
+
+This is internal package documentation describing the vectorized formulation of the Convex Clustering problem, CARP and CBASS.
+
+The goal is describe how CARP and CBASS make their updates at each iteration, and also describe some of the auxillary functions
+which are used to accomplish this.
+
+## Original Problem
+
+The Convex Clustering problem is given by
+
+$$
+\min_{\boldsymbol{U}} \frac{1}{2} \| \boldsymbol{X} - \boldsymbol{U} \|_F^2 + \lambda \sum_{(i,j) \in E} w_{ij} \| \boldsymbol{u}_{i} - \boldsymbol{u}_{j} \|_2
+$$
+here
+
+* $\boldsymbol{X} \in \mathbb{R}^{p \times n}$ is our data matrix consisting of $n$ observations in $p$ dimensions
```

Comment:
This is worth thinking about (in light of #45). What changes if we work with the "real" `X` instead of its transpose?